### PR TITLE
Activation and rejection mail response

### PR DESCRIPTION
**Pull request by ahmedannahir**

### CODE REVIEW

File: src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java [hunk 1]
```diff
@@ -14,7 +14,5 @@ public interface ClientProfileDAO extends JpaRepository<ClientProfile, Long> {
 
     List<ClientProfile> findClientProfileByAccountStatus(AccountStatus inactive);
 
-    ClientProfile findClientProfileById(long id);
```
Comment (ArouayAb) [inline src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java:17]: Why is this removed?

File: src/main/java/ensa/ebanking/accountservice/Services/AgentService.java [hunk 2]
```diff
@@ -57,24 +62,50 @@ public List<ClientProfile> invalideAccounts () {
...
+                return clientProfileDAO.save(client);
+            }
+            else {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+            }
 
         } catch (Exception e) {
-            System.out.println("account not found");
+            e.printStackTrace();
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (ArouayAb) [inline src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: Internal Server Error doesn't suit the purpose here, 500 means an unexpected error has happened.

File: src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java (hunk 1 above)
Comment (ahmedannahir) [reply src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java:17]: Worked with one already there in spring data that returns Optional<ClientProfile> which returns a ClientProfile if there's one in the db and a null if there isn't.

I haven't tested the app with "ClientProfile findClientProfileById(long id);", but the other one works and this one wasn't used anywhere else so it being removed doesn't affect anything.

File: src/main/java/ensa/ebanking/accountservice/Services/AgentService.java (hunk 2 above)
Comment (ahmedannahir) [reply src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: Which error do you suggest?

File: src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java (hunk 1 above)
Comment (ArouayAb) [reply src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java:17]: The request comes with an id not with the entire ClientProfile. I see that you changed it. Does bader know about this ?

File: src/main/java/ensa/ebanking/accountservice/Services/AgentService.java (hunk 2 above)
Comment (ArouayAb) [reply src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: I suggest 404 not found

File: src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java (hunk 1 above)
Comment (ahmedannahir) [reply src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java:17]: No, I didn't tell him about this. The method parameter was a String Json and I thought it had a ClientProfile in the body as a json. I can turn this back to just an id in the body, or I can tell Bader to send ClientProfile. Whichever you choose.

File: src/main/java/ensa/ebanking/accountservice/Services/AgentService.java (hunk 2 above)
Comment (ahmedannahir) [reply src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: I sent that one when the Optional<ClientProfile> is null, i.e there's no user in the database with that id. This one was when any other error was raised. I can make this a 404 error if you want.

Comment (ArouayAb) [reply src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: fga3 l a7wal 404 hsn mn 500

Comment (ahmedannahir) [reply src/main/java/ensa/ebanking/accountservice/Services/AgentService.java:84]: Ok will change it and make another pull request

File: src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java (hunk 1 above)
Comment (ArouayAb) [reply src/main/java/ensa/ebanking/accountservice/DAO/ClientProfileDAO.java:17]: Do whichever you want, either talk with him and leave it as clientprofile or revert it to id.
